localstack/services/moto.py

```diff
@@ -65,7 +69,18 @@ def call_moto_with_request(
         region=context.region,
     )
 
-    local_context.request.headers.update(context.request.headers)
+    if override_headers:
+        # by default, we first re-serialize the parameters from the `service_request` and then we update the
+        # local_context headers with the ones from the original request, which means you will override your previously
+        # set parameter with the one from the original request.
+        # we reverse the order if `override_headers` is set, and properly override the original request with all
+        # parameters from the `service_request`
+        headers = copy.deepcopy(context.request.headers)
+        headers.update(local_context.request.headers)
+        local_context.request.headers = headers
```

Comment:
This solution is looking way better. If this works fine, I think this should actually just be the default?

Reply:
Yes, I don't know what happened but I got carried away 😅 thanks for catching that. I'll try a test run with this, you're right, it's the correct behaviour!